pkg/errors/errors.go

```diff
@@ -0,0 +1,35 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+)
+
+type Error struct {
```

Comment:
Now that we have this refactored out we should add some more context and explanation here. 

Some things that stand out based on current usage:

- There seems to be the implicit expectation that all Errors are used with the `Wrap` function in order to return an `error` type. 
- The Params and Message are linked such that the message is defined initially with an assumed param which must then be known and provided when used. 
- Error() is never used. Is it defined to satisfy the errors interface? If so, then why does Wrap return error and not Error?

Reply:
Well using `Wrap` is an option since this `Error` struct implements `errors.Error`, it can be simply casted implicitly to and `error` type.  But good point, to make it unified and more clear we can make the `Wrap` to return an `*Error` type instead. which can be even more handy for the users.

and yes `Error()` is implemented to satisfy the error interface and it is called when an error gets logged or printed out.

Reply:
done